src/canvas-extractor/edfi_canvas_extractor/extract_graphql.py

```diff
@@ -126,6 +137,26 @@ def _write_sections_associations(arguments: MainArguments) -> None:
     )
 
 
+def _get_assignments(
+    arguments: MainArguments,
+    gql: GraphQLExtractor,
+    sync_db: sqlalchemy.engine.base.Engine,
+) -> bool:
+    logger.info("Extracting Assignments from Canvas")
+    (sections, _, all_section_ids) = results_store["sections"]
+    sections_df = to_df(sections)
+    (assignments, udm_assignments_df) = extract_assignments(
+        sections_df, gql, sync_db
+    )
+    logger.info("Writing LMS UDM Assignments to CSV files")
+    write_assignments(
```

Comment:
SUGGESTION: This can be separated out into it's own function `_write_assignments` with the `@catch_exceptions` decorator like we have for other similar operations in the file.